File: examples/python/Misc/Bootloader/bootloader_dump.py
#!/usr/bin/env python3

import depthai as dai


found, deviceInfo = dai.DeviceBootloader.getFirstAvailableDevice()
if not found:
    raise RuntimeError("No available device found")

deviceState = str(deviceInfo.state).replace("XLinkDeviceState.", "")

with dai.DeviceBootloader(deviceInfo) as bootloader:
    version = bootloader.getVersion()
    userBootloader = bootloader.isUserBootloader()

bootloaderType = "Factory flashed bootloader" if not userBootloader else "User flashed bootloader"

print(deviceInfo)
print(f"Bootloader version: {bootloaderType} {version}")
print(f"Embedded depthai bootloader version: {dai.DeviceBootloader.getEmbeddedBootloaderVersion()}")

File: examples/cpp/Misc/Bootloader/bootloader_dump.cpp
#include <iostream>
#include <stdexcept>
#include <string>
#include <tuple>

#include "depthai/depthai.hpp"

int main() {
    bool found = false;
    dai::DeviceInfo deviceInfo;
    std::tie(found, deviceInfo) = dai::DeviceBootloader::getFirstAvailableDevice();
    if(!found) {
        throw std::runtime_error("No available device found");
    }

    std::string version;
    bool userBootloader = false;
    {
        dai::DeviceBootloader bootloader(deviceInfo);
        version = bootloader.getVersion().toString();
        userBootloader = bootloader.isUserBootloader();
    }

    const std::string bootloaderType = userBootloader ? "User flashed bootloader" : "Factory flashed bootloader";

    std::cout << deviceInfo.toString() << '\n';
    std::cout << "Bootloader version: " << bootloaderType << ' ' << version << '\n';
    std::cout << "Embedded depthai bootloader version: " << dai::DeviceBootloader::getEmbeddedBootloaderVersion() << '\n';
    return 0;
}
